docs/architecture/incremental.md

```diff
@@ -0,0 +1,22 @@
+# Configuration: Streams & Incremental Syncs
+
+## AirbyteStream
+* The `AirbyteStream` has two sections:
+    * The `discovered` section describes information and metadata about the data in the source. This section is generated by the source connector. Outside of the source, this section should be treated as read only.
+    * The `configuration` section describes functionality that an end user would like applied on the sync. This section is configured by the end-user, and is never touched by the source.
+
+## Discovered
+
+Currently this section contains two fields: `json_schema` and `metadata`.
+* `json_schema` describes the structure of the data in a source.
+* `metadata` describes other information about the data in the source. For example what features are supported for the stream. Currently the `metadata` object contains two fields.
+    * `supported_sync_modes` which describes which sync modes are supported for a stream.
+    * `default_comparable` which describes what field should be used to determine if a record should be included in a sync. If this is set, the end user does not need to specify their own comparable. If it is not set, then the end user does need to specify it. If it is set and the user sets their own, the user's value overrides the default value.
+
+## Configuration
+* This section currently has two fields: `sync_mode` and `comparable`.
+* `sync_mode` which describes how the data will be synced (e.g. `full_refresh` or `incremental`).
+* `comparable` which is the path to the field that will be used to determine if a record is new or modified since the last sync. This field is REQUIRED if `sync_mode` is `incremental`. Otherwise it is ignored.
+
+## Source
+A source advertises that it can do an incremental sync for a given stream, by specifying `incremental` as one of the sync modes. It can specify a `default_comparable` so that the user does not need to determine which field is used to determine which records are new or modified. If it does not specify this field, then the user _must_ set `comparable` field themselves if they want to use `incremental` syncing.
```

Comment:
Should we add some details on what `incremental` means for Airbyte at the moment?

ie we support key-based incremental replications via comparable columns not log-based or CDC incremental modes which wouldn't need comparable columns but not in the scope for the moment.

Reply:
will remove docs for now. causing confusion as we iterate on the spec.